matrix/Matrix.hpp

```diff
@@ -24,28 +24,28 @@ namespace matrix
 template <typename Type, size_t M>
 class Vector;
 
-template<typename Type, size_t  M, size_t N>
+template<typename Type, size_t M, size_t N>
 class Matrix
 {
 
 public:
 
-    Type _data[M][N];
+    Type _data[M][N] {};
 
     // Constructors
-    Matrix() : _data() {}
+    Matrix() = default;
 
-    Matrix(const Type data_[][N]) : _data()
+    Matrix(const Type data_[M])
```

Comment:
Do we need this one? If N!=1 you'll have an illegal memory access. Changing to `sizeof(data_)` would be fine too.

Reply:
We do for now, a lot of the testing constructs a matrix from an array. Check the updated commit.